src/CurrentRoute/Factory/View/Helper/CurrentRouteFactory.php

```diff
@@ -2,21 +2,30 @@
 
 namespace CurrentRoute\Factory\View\Helper;
 
-use Zend\ServiceManager\FactoryInterface;
+use Interop\Container\ContainerInterface;
+use Zend\ServiceManager\Factory\FactoryInterface;
 use Zend\ServiceManager\ServiceLocatorInterface;
-use Zend\Mvc\Router\Http\RouteMatch;
 use \CurrentRoute\View\Helper\CurrentRoute;
 
 class CurrentRouteFactory implements FactoryInterface
 {
+
+    /** 
+     * Let's return view helper interface
+     */
+     
+     public function __invoke(ContainerInterface $container, $requestedName, array $options = null )
+     {
+         return $this->createService($container);
+     }
     /**
      * @param ServiceLocatorInterface $serviceLocator
      *
      * @return CurrentRoute
      */
     public function createService(ServiceLocatorInterface $serviceLocator)
     {
-        $routeMatch = $serviceLocator->getServiceLocator()->get('Application')->getMvcEvent()->getRouteMatch();
```

Comment:
The PR is really nice idea but unfortunately this does not works :(
since you can not use  get() without getServiceLocator()